src/transformers/generation/utils.py

```diff
@@ -4923,3 +4994,106 @@ def _ranking_fast(
     contrastive_score = torch.stack(torch.split(contrastive_score, beam_width))  # [B, K]
     _, selected_idx = contrastive_score.max(dim=-1)  # [B]
     return selected_idx
+
+
+def _split_model_inputs(
+    obj: Union[ModelOutput, Dict], split_size: int, full_batch_size: int
+) -> List[Union[ModelOutput, Dict]]:
+    """
+    Split a ModelOutput object (or its subclasses) or Dict into a list of same-class objects based on a specified split
+    size. The input object is dict when it was prepared for forward pass and ModelOutput when it was returned from
+    previous forward pass.
+    """
+    # Infer the class from the object
+    cls = type(obj)
+    assert (full_batch_size % split_size) == 0, "`full_batch_size` must be divisible by `split_size`"
```

Comment:
We avoid using `assert` in `transformers`, raising the appropriate exception instead (e.g. `ValueError`)